docs/docs/why-gatsby-uses-graphql.md

```diff
@@ -0,0 +1,360 @@
+
+The results that come back are very similar to the contents of `data/products.json`, so you can loop through the results and create a page for each.
+
+However, note that you’re only passing the `slug` in `context` — you’ll use this in the template component to load more product data.
+
+As you’ve already seen, the `context` argument is made available to the template component in the `pageContext` prop. To make queries more powerful, Gatsby _also_ exposes everything in `context` as a GraphQL variable, which means you can write a query that says, in plain English, “Load data for the product with the slug passed in `context`.”
+
+Here’s what that looks like in practice:
+
+```jsx:title=src/templates/product-graphql.js
+import React from "react"
+import { graphql } from "gatsby"
+import Image from "gatsby-image"
+
+export const query = graphql`
+  query($slug: String!) {
+    productsJson(slug: { eq: $slug }) {
+      title
+      description
+      price
+      image {
+        childImageSharp {
+          fluid {
+            ...GatsbyImageSharpFluid
+          }
+        }
+      }
+    }
+  }
+`
+
+const Product = ({ data }) => {
+  const product = data.productsJson
+
+  return (
+    <div>
+      <h1>{product.title}</h1>
+      <Image
+        fluid={product.image.childImageSharp.fluid}
+        alt={product.title}
+        style={{ float: "left", marginRight: "1rem", width: 150 }}
+      />
+      <p>{product.price}</p>
+      <div dangerouslySetInnerHTML={{ __html: product.description }} />
+    </div>
+  )
+}
+
+export default Product
+```
+
+A few notes about this file:
+
+1. The result of the query is added to the template component as the `data` prop.
+2. The image path was automatically converted by the Sharp transformer into a “child node” that includes optimized versions of the image.
+3. The query uses a [GraphQL fragment](https://www.gatsbyjs.org/packages/gatsby-image/#fragments) to query all the required data for optimized images. GraphQL fragments _do not work_ in the GraphQL Playground.
+4. The `img` tag has been swapped out for a `gatsby-image` component named `Image`. Instead of a `src` attribute, it accepts an object with optimized image data.
+
+Save this file, run `gatsby develop`, then open `localhost:8000/gql/purple-hat/`:
+
+![Lazy loaded image of an angry cat wearing the purple hat.](./images/why-gql-images.gif)
+
+The image is now optimized and lazy loaded.
+
+After the initial setup, loading data with GraphQL is fairly similar to directly loading JSON, but it provides extra benefits like automatically optimizing images and keeping the data loading in the same place where it’s used.
+
+It’s certainly not required, and in simple use cases it might not be worth the effort, but in many sites and apps, GraphQL will vastly simplify the process of building and optimizing your pages.
```

Comment:
I'd maybe again just tweak this to be stronger. We can and should hold that GraphQL is best practice for structuring Gatsby applications.

```suggestion
GraphQL is certainly not required--as shown earlier--but the benefits of adopting GraphQL are obvious and numerous. GraphQL will generally simplify the process of building and optimizing your pages, and so it is and will remain a best practice in structuring and writing your next, great Gatsby application.
```

Reply:
Made some tweaks here, but I like it!